src/pages/DailyGardening/index.tsx

```diff
@@ -140,8 +131,40 @@ export const DailyGardening = () => {
     [userData]
   );
 
+  const removeCompletedTask = useCallback(
+    async (rule: Rule) => {
+      if (rule._id) {
+        setLastClicked(rule._id);
+      }
+
+      const taskToDelete = currentCompletedTasks.find(
+        (completedTask) =>
+          completedTask.ruleId === rule._id &&
+          isSameDay(new Date(), new Date(completedTask.date))
+      );
+
+      if (taskToDelete && userId) {
+        // will return updated coins for users
+        await deleteTask(taskToDelete._id, userId);
+      }
+    },
+    [currentCompletedTasks, deleteTask, userId]
+  );
+
+  const onTaskClicked = useCallback(
+    async (rule: Rule, isRuleCompleted: boolean) => {
+      if (!isRuleCompleted) {
+        completeTaskHandler(rule);
+      } else {
+        removeCompletedTask(rule);
+      }
+    },
+    [completeTaskHandler, removeCompletedTask]
+  );
+
   useEffect(() => {
     if (completedTaskApi.status === "succeeded") {
+      // MEMO: Return also deleted task id and not just user account data
       const { balance: newCoinBalance } = completedTaskApi.response.user;
       setCurrentCompletedTasks((v) => [
```

Comment:
Currently api return s only user account updated data, would be nice to have the deleted task data too at least the id

Reply:
We can fix this on the backend after the demo.